### DISCUSSION TITLE
Annotations disappearing

### DISCUSSION THREAD

**Discussion by javisin22** [sumatrapdfreader/sumatrapdf#3926] [Q&A]
I often annotate a pdf file using the underline, highlight or create text annotation under cursor. Every time I use one of these, I perform the shortcut "CTRL + SHIFT + S"  in order to save the annotations to existing pdf. The problem is that once I annotate the same file several times the previous annotations start disappearing and don't show, although the file's size increases as if the annotations are still somewhere hidden in the file, and if I open the file in other viewers such us Chrome or Edge browsers ALL the annotations appear.
I've tried in Adobe Acrobat PDF and annotations don't show as in Sumatra.

What is happening? How can I do in order to see the full anotation set of the file in Sumatra?

I have the file stored in the cloud, exactly in OneDrive, and I've tried downloading the file to use it locally and it doesn't work.

**Comment by GitHubRulesOK** [comment]
Each time you save it rewrites the file with additional amendments that duplicates parts of the file references, thus can bloat if saved for each letter / line highlighted object. So it is best to wait and save after a page worth of additions  or a timed interval. Also each time you "delete" the old edits are retained (or should be) just flagged as dont show.

Clearly the more often you save the greater risk of corrupting the tail of file, but the longer you wait to save the greater the lost time per fail.

What can interfere with an editor is
 1) AntiVirus, less common nowadays on say PDF but they are still candidates for scanning whilst changing a file contents.
 2) a folder synchronisation app that does not expect a PDF data file to be modified so will insist on tugging the rug.

How to avoid such problems is 
1) only edit in a local data area that is not managed by protection or synchronisation. So use c:\my data folder\ rather than my c:\users\me\one somewhere else drive\desktop
2) mark that workzone as not needing monitoring by AV or sync

**Reply by protaniv** [reply]
How come the Edge browser show all annotations while Adobe Acrobat and Sumatra don't. Firefox also doesn't show all annotations. 

The only thing that works with the filesystem is the built in Windows 10 file explorer. Happens even when the folder is not open. The  Avast AV has an exception for scanning for that particular folder. Even with disabling the AV it happens. And it is consistent. For even 1 or 2 lines of annotations it will "forget" to show the previous ones.

The funny this is this.
If I edit the file with my Edge browser which has a bit different color of the annotation mark, it will SHOW ALL annotation on Adobe Acrobat, Sumatra AND Firefox. It will also remember all consequent annotations.

If i save the annotations on a file that has been changed only with Sumatra, and then try to save annotations in Adobe Acrobat. The annotations disappear but then even Edge can't see them anymore.

After that Sumatra annotates without issues. Feels like the new annotation from edge forces it to show all annotations. For both Sumatra and Adobe.

**Reply by GitHubRulesOK** [reply]
@protaniv your covering many differences. When Edge uses annotations it does not use the same method as SumatraPDF and many other PDF readers (where the change is "incremental" i.e. appending to end of an existing file) it rebuilds the Whole PDF. Thus in effect burning any comments into the core of the PDF (more permanent).

There have been various issues with annotation while under development, such that only the current pre-release (today it is 3.6.15915 but not correctly on offer only yesterdays 3.6.15914 with 1 known bug) can be used as a yardstick for testing current mis behaviours. However while editing a file it must be under local NTFS/MFT disk control not controlled as a virtual drive).

**Reply by protaniv** [reply]
@GitHubRulesOK  Ok, thanks for the reply. I just wanted to share an experience I have as it might be of some help to explain the behavior so I tried to be precise. This is the main point which fixes the issue for me and I thought it might be interesting:

"...If I edit (and save) the file with my Edge browser ... it will SHOW ALL annotation...
...After that Sumatra annotates without issues. Feels like the new annotation from edge forces it to show all annotations..."

Best regards.

**Reply by protaniv** [reply]
Ok, thank you for explaining. I just wanted to share an interesting behavior that might be of interest or helpful.

The main point I wanted to describe is this:
"If I edit the file with my Edge browser ... it will SHOW ALL annotation on Adobe Acrobat, Sumatra AND Firefox. It will also remember all consequent annotations."

Thank you for your time. Have a nice day.

**Reply by Tomas-Bueno-Momcilovic** [reply]
The same issue persists in v3.5.2. This makes it difficult to use SumatraPDF for any serious reviews.

For debugging purposes, here's how you can reproduce this:

0. Precondition: PDF in a folder synchronized with OneDrive. 
1. Place an annotation somewhere;
2. Save it using CTRL+SHIFT+S;
3. Place another annotation;
4. Click on "Save in current document" button in the pop-up annotations window.

This effectively nukes all existing annotations.

**Reply by GitHubRulesOK** [reply]
@Tomas-Bueno-Momcilovic there is NEVER a means to use Adobe edits in one drive per its design 
One drives never existed in 1995 it is a failure in the Adobe methodology that Synchronised drives could never be designed to guard against

It is not a bug in PDF it is a bug in PARTIAL editing files in a parts synchronised system.
NEVER edit Word documents in One Drive without they get locked in a MS managed way.
I have lost many files and directories in Office Environments nothing to do with PDF so I never use One Drive (nor similar when editing)

**Reply by kjk** [reply]
@Tomas-Bueno-Momcilovic 3.5.2 has known issues with saving annotations in some cases.

Please upgrade to latest pre-release https://www.sumatrapdfreader.org/prerelease

If you still have issues there, please open a new issue with detailed instructions to reproduce the problem.

As far as I know there are no issues with pre-release.
